addons/address_autocomplete/static/src/js/address_autocomplete.js

```diff
@@ -0,0 +1,82 @@
+/** @odoo-module **/
+
+import { ExtendedAutocomplete } from "./extended_autocomplete";
+import { registry } from "@web/core/registry";
+import { rpc } from "@web/core/network/rpc";
+import { _t } from "@web/core/l10n/translation";
+import { CharField, charField } from "@web/views/fields/char/char_field";
+
+export class AddressAutoComplete extends CharField {
+    static template = "address_autocomplete.AddressAutoCompleteTemplate";
+    static components = { ExtendedAutocomplete, ...CharField };
+
+    setup() {
+        this.sessionId = this._generateUUID();
+    }
+
+    /**
+     * Used to generate a unique session ID for the places API.
+     *
+     * @private
+     */
+    _generateUUID() {
+        return "xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx".replace(/[xy]/g, function (c) {
+            const r = (Math.random() * 16) | 0,
+                v = c == "x" ? r : (r & 0x3) | 0x8;
+            return v.toString(16);
+        });
+    }
+
+    get sources() {
+        return [
+            {
+                options: async (request) => {
+                    if (request.length > 5) {
+                        const suggestions = await rpc("/autocomplete/address", {
+                            partial_address: request,
+                            session_id: this.sessionId || null,
+                        });
+                        return suggestions.results;
+                    } else {
+                        return [];
+                    }
+                },
+                optionTemplate: "address_autocomplete.CharFieldDropdownOption",
+                placeholder: _t("Searching for addresses..."),
+            },
+        ];
+    }
+
+    async onSelect(ev) {
+        const address = await rpc("/autocomplete/address_full", {
+            address: ev.formatted_address,
+            google_place_id: ev.google_place_id,
+            session_id: this.sessionId || null,
+        });
+
+        const dict = { [this.props.name]: address.formatted_street_number || "" };
+        const fieldsToUpdate = ["street2", "city", "zip", "country_id", "state_id"];
+
+        fieldsToUpdate.forEach((field) => {
+            if (this.env.model.config.fields[field]) {
```

Comment:
> @aab-odoo In theory, do you think that a widget could update multiple other fields's data at the same time ?

Yes, it is supported

Reply:
> wrongly accessing fields: it should be accessed via record, not model.

Done

> If it is not generic and only applies to res.partner-like models, then we should have a way to prevent its use in other contexts.

Yes, this applies to modules like res_partner, res_company,etc.

> this field seems very coupled to how the model is declared. What if a model wants to use that, but zip is called "postal_code" ?

I need to update the address_fields when selecting options from the dropdown. I noticed that in res_partner, the address_fields are declared in the same way. As you mentioned, field names can sometimes differ. In this case, how can we identify whether a field is related to an address or not?
https://github.com/odoo/odoo/blob/b3d78b5b5c28a09ebfd6a6b4f5130b312fe0821c/odoo/addons/base/models/res_partner.py#L37